validation-test/compiler_crashers_2_fixed/issue-55435.swift

```diff
@@ -6,8 +6,7 @@ class ContainerTransition {
   var viewControllers: [Int: String]?
   func completeTransition() {
     viewControllers?[Int//.max
-    // expected-error@-1 {{no exact matches in call to subscript}}
-    // expected-note@-2 {{found candidate with type '(Int.Type) -> Dictionary<Int, String>.SubSequence' (aka '(Int.Type) -> Slice<Dictionary<Int, String>>')}}
-    // expected-note@-3 {{to match this opening '['}}
+    // expected-error@-1 {{cannot convert value of type 'Int.Type' to expected argument type 'Int'}}
+    // expected-note@-2 {{to match this opening '['}}
```

Comment:
This is because previously the parser didn't wrap the expression with `OptionalEvaluationExpr` if it has an parse error.